daemon/tests/happy_path.rs

```diff
@@ -191,6 +191,43 @@ async fn force_close_an_open_cfd() {
     assert_next_state!(CfdState::Closed, maker, taker, order_id);
 }
 
+#[tokio::test]
+async fn rollover_an_open_cfd() {
+    let _guard = init_tracing();
+    let oracle_data = OliviaData::example_0();
+    let (mut maker, mut taker, order_id) =
+        start_from_open_cfd_state(oracle_data.announcement()).await;
+
+    taker.trigger_rollover(order_id).await;
+
+    let (taker_cfd, maker_cfd) = next_cfd(taker.cfd_feed(), maker.cfd_feed()).await.unwrap();
+    assert_eq!(taker_cfd.state, CfdState::OutgoingRollOverProposal);
+    assert_eq!(maker_cfd.state, CfdState::IncomingRollOverProposal);
```

Comment:
With a bit more macro trickery to accept different syntaxes, we should be able to create a version of `assert_next_state` that allows us to assert on different events for maker / taker.

See https://doc.rust-lang.org/rust-by-example/macros/overload.html for example.

Reply:
I briefly added something like what you suggest, but less elegant. I'll take a look at that link!